api/client/src/main/java/org/projectnessie/client/http/HttpClientBuilderImpl.java

```diff
@@ -240,32 +252,57 @@ public HttpClient build() {
             .addAllResponseFilters(responseFilters)
             .isHttp11Only(!http2Upgrade)
             .followRedirects(followRedirects)
-            .forceUrlConnectionClient(forceUrlConnectionClient)
             .authentication(authentication)
             .build();
 
-    return ImplSwitch.FACTORY.apply(config);
+    String clientName = httpClientName;
+    // "HTTP" is the name for `org.projectnessie.client.http.NessieHttpClientBuilderImpl`
+    if ("HTTP".equalsIgnoreCase(clientName)) {
+      // fall back to default
+      clientName = null;
+    }
+    if (clientName == null && forceUrlConnectionClient) {
+      // forced use of URLConnectionClient
+      clientName = "URLConnection";
+    }
+    if (clientName != null
+        && forceUrlConnectionClient
+        && !"URLConnection".equalsIgnoreCase(clientName)) {
+      throw new IllegalArgumentException(
```

Comment:
Is this validation worth the trouble? Why not let `forceUrlConnectionClient` with a WARN log message? The intent of that property is clear. On the other hand, breaking on misalignment with client name may complicate rollouts in case config is shared across multiple versions.